pkg/agent/secondarynetwork/podwatch/controller.go

```diff
@@ -114,6 +124,53 @@ func podKeyGet(pod *corev1.Pod) string {
 	return pod.Namespace + "/" + pod.Name
 }
 
+// buildVFDeviceIDListPerPod is a helper function to build a cache structure with the
+// list of all the PCI addresses allocated per Pod based on their resource requests (in Pod spec).
+// When there is a request for a VF resource (to associate it for a secondary network interface),
+// getUnusedSriovVFDeviceIDPerPod will use this cache information to pick up a unique PCI address
+// which is still not associated with a network device name.
+// NOTE: buildVFDeviceIDListPerPod is called only if a Pod specific VF to Interface mapping cache
+// was not build earlier. Sample initial entry per Pod: "{18:01.1,""},{18:01.2,""},{18:01.3,""}"
+func (pc *PodController) buildVFDeviceIDListPerPod(podName, podNamespace string) error {
+	key := podNamespace + "/" + podName
+	pc.vfDeviceIDCacheMutex.RLock()
+	defer pc.vfDeviceIDCacheMutex.RUnlock()
```

Comment:
To shorten the lock period, we can unlock after reading the map, and lock (with write lock) again when updating the map.